src/PostgREST/App.hs

```diff
@@ -419,6 +419,15 @@ mutateRequest apiRequest = mapLeft (errResponse status400) $
       case target of
         (TargetIdent (QualifiedIdentifier _ t) ) -> t
         _ -> undefined
+    fieldNames :: ReadRequest -> PreferRepresentation -> [FieldName]
+    fieldNames _ None = []
+    fieldNames (Node (Select colSelects _ _ _ _, (_, _, _)) forest) _ =
```

Comment:
There are a lot of underscores, what about matching `(Node sel forest)` and then using the `select` accessor from Select to obtain what you call `colSelects` here?